opensearch_py_ml/ml_models/sparse_model.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+import os
+from zipfile import ZipFile
+
+import requests
+
+LICENSE_URL = "https://github.com/opensearch-project/opensearch-py-ml/raw/main/LICENSE"
+
+
+class SparseModel:
+    """
+    Class for autotracing the Sparse Encoding model.
+    """
```

Comment:
As sentence transformer is also a kind of model, have you considered implementing a base class for ml_models?

Reply:
+1